### DISCUSSION TITLE
How to convert an integer field to a Relationship?

### DISCUSSION THREAD

**Discussion by caschbre** [directus/directus#6320] [Archive]
I've spun up directus on an existing project. The current mysql schema does not contain foreign key information. Primary keys tend to be auto-increment integers and "foreign" key columns are simple integers.  E.g. posts.uid -> users.uid.  I would like to alter the posts.uid field on the Posts collection to be a relationship. However when editing the field, it's locked as an integer.  

How would I go about establishing the relationship between Posts and Users without creating a new field?

**Comment by rijkvanzanten** [comment]
> The current mysql schema does not contain foreign key information.

This is the key part here (pun intended). Directus will rely on the database schema itself to determine what types to use. If you add the foreign key constraint in the database directly, it will work as expected in Directus 👍🏻

**Reply by caschbre** [reply]
Yeah, I love that it reads the current schema and sets up the collections to match.  Unfortunately we're not able to setup the foreign keys in the db given how the app runs... which is why I was hoping to be able to setup the relationship in directus without requiring the schema change.  I'm assuming from your answer that this isn't feasible?

**Reply by rijkvanzanten** [reply]
I see, in that case, there's another thing you can do: use the `/relations` endpoint directly (https://docs.directus.io/reference/api/system/relations/#create-a-relation)

You should be able to do a POST request as follows:

```json
// POST /relations

{
	"collection": "users",
	"field": "posts_uid",
	"related_collection": "posts"
}
```

(i'm assuming that your foreign key column is `users.posts_uid` which stores `posts.uid` (primary key). Update those as necessary)

**Reply by philleepflorence** [reply]
@rijkvanzanten - is this a plausible work around for PK INT to PK UUID?

| Posts | Posts Users |Directus Users |
| ----------- | ----------- | ----------- |
| user INT | post user  | - |
| - | directus user  | UUID |

...saw your suggestion right before submitting!

**Reply by rijkvanzanten** [reply]
What you're describing here is a regular many to many setup 🙂

**Reply by philleepflorence** [reply]
Ah, [Create a Relation](https://docs.directus.io/reference/api/system/relations/#create-a-relation) allows One-2-One Relationship!?
Awesome!

**Reply by rijkvanzanten** [reply]
Not officially no. I suppose you could setup two many to ones where the FK is also the PK so you get an id<->id relationship, which enforces the one to one, but haven't tried that 🤔

**Reply by caschbre** [reply]
> I see, in that case, there's another thing you can do: use the `/relations` endpoint directly (https://docs.directus.io/reference/api/system/relations/#create-a-relation)
> 
> You should be able to do a POST request as follows:
> 
> ```json
> // POST /relations
> 
> {
> 	"collection": "users",
> 	"field": "posts_uid",
> 	"related_collection": "posts"
> }
> ```
> 
> (i'm assuming that your foreign key column is `users.posts_uid` which stores `posts.uid` (primary key). Update those as necessary)

Ah ok.   So if I want the relationship the other way around... posts.uid --> users.uid then I would do the following?
```
  {
    "collection": "posts",
    "field": "uid",
    "related_collection": "users",
  }
```

**Reply by rijkvanzanten** [reply]
> posts.uid --> users.uid

Just to be absolutely clear, is `users.uid` the _foreign_ key field to `posts.uid`? You wouldn't want the `posts.uid` to overwrite the `users.uid`. I would've assumed your structure to be something like

```
posts
--------------
uid (primary key)
author (foreign key to users)
```

```
users
-------------
uid (primary key)
favorite_post (foreign key to posts)
```

**Reply by philleepflorence** [reply]
> ```json
> // POST /relations
> 
> {
> 	"collection": "users",
> 	"field": "posts_uid",
> 	"related_collection": "posts"
> }
> ```
> 
> (i'm assuming that your foreign key column is `users.posts_uid` which stores `posts.uid` (primary key). Update those as necessary)

Sorry...momentary dyslexia! Thought I saw `related_field`.

**Reply by caschbre** [reply]
users
---------
uid (pk)

posts
--------
id: (pk)
uid: (fk)

The uid is always our user primary key and is often used on other tables to represent the author... so it's a foreign key on those other tables.

**Reply by rijkvanzanten** [reply]
Ohhh gotcha! I thought "uid" stood for "unique identifier" instead of "user id" 🙂

I'm that case, your previous example is correct 👍🏻

**Reply by caschbre** [reply]
Awesome, thanks!  I think that's the last major hurdle for me.
